src/ec2_metadata/__init__.py

```diff
@@ -97,6 +97,19 @@ def iam_info(self) -> Optional[Dict[str, Any]]:
             return None
         return resp.json()
 
+    @cached_property
+    def iam_security_credentials(self) -> Optional[Dict[str, Any]]:
+        instance_profile_name = self.instance_profile_name
+        if instance_profile_name is None:
+            return None
+        resp = self._get_url(
+            METADATA_URL + "iam/security-credentials/" + instance_profile_name,
+            allow_404=True,
+        )
+        if resp.status_code == 404:
+            return None
```

Comment:
isn't there a bug if the credentials don't exist?

Reply:
Can you expand? I never encountered this case, is it possible to have an IAM profile but no credentials?

Reply:
I mean, if `instance_profile_name` is not `None`, then a 404 here would seem to be an error, for which we should raise (?) rather than silence by returning `None`.

Reply:
I see, yes. Will fix.